File: practice-problems/twosum/twosum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indexDict = { nums[i]:i for i in range(len(nums))}
        
        for i in range(len(nums)):
            curr = nums[i]
            needed = target - curr
            if needed in indexDict and indexDict[needed] != i:
                return [i, indexDict[needed]]
    
    
nums = [2,7,11,15]
target = 9

s = Solution()

print(s.twoSum(nums, target))

File: practice-problems/twosum/twosum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> indexMap;
            for(int i= 0; i< nums.size(); i++){
                indexMap[nums[i]]=i;
            }

            for(int j = 0; j < nums.size(); j++){
                int curr = nums[j];
                int needed = target-curr;
                if(indexMap.count(needed) > 0 and indexMap[needed] != j){
                    return {j,indexMap[needed]};
                }
            }
            return {-1,-1};

        }
    };

int main() {
        Solution sol;
    
        vector<int> nums = {2, 7, 11, 15};
        int target = 9;
    
        vector<int> result = sol.twoSum(nums, target);
    
        cout << "Indices: ";
        for (int index : result) {
            cout << index << " ";
        }
        cout << endl;
    
        return 0;
    }